cmd/clusterctl/cmd/rollout/pause.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rollout
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+)
+
+// pauseOptions is the start of the data required to perform the operation.
+type pauseOptions struct {
+	kubeconfig        string
+	kubeconfigContext string
+	resources         []string
+	namespace         string
+}
+
+var pauseOpt = &pauseOptions{}
+
+var (
+	pauseLong = templates.LongDesc(`
+		Mark the provided cluster-api resource as paused.
+
+	        Paused resources will not be reconciled by a controller. Use "clusterctl alpha rollout resume" to resume a paused resource. Currently only MachineDeployments support being paused.`)
+
+	pauseExample = templates.Examples(`
+		# Mark the machinedeployment as paused.
+		clusterctl alpha rollout pause machinedeployment/my-md-0
+`)
+)
+
+// NewCmdRolloutPause returns a Command instance for 'rollout pause' sub command
+func NewCmdRolloutPause(cfgFile string) *cobra.Command {
```

Comment:
are you planning to add docs for the pause/resume command in a separate PR?

Reply:
I actually don't see any docs for the rollout command in https://github.com/kubernetes-sigs/cluster-api/tree/master/docs/book/src/clusterctl/commands, are we tracking that in an issue?

Reply:
@CecileRobertMichon There is a tracking issue and docs is in the list: #3439
If it's okay with you, I want to wait for a few more rollout subcommands to merge before opening a PR for the docs.